pyccel/stdlib/lists/lists.c

```diff
@@ -0,0 +1,665 @@
+        Parameters
+        ----------
+        list : t_list**
+            a pointer on the list to be freed
+*/
+{
+    list->size = 0;
+}
+
+t_list*  copy(t_list* list)
+/*
+        Creates a copy t_list object from another
+
+        Parameters
+        ----------
+        list : t_list*
+            a pointer on the list to be copied
+        
+        Returns
+        =======
+        list : t_list
+*/
+{
+    return (allocate_list(list->size, list->type, list->elements));
+}
+
+size_t   count(t_list* list, void *item)
+/*
+        Counts the number of occurrences of a specific element
+
+        Parameters
+        ----------
+        list : t_list*
+            the t_list where to look for occurrences
+
+        item : void*
+            The element to look for
+
+        Returns
+        =======
+        count : size_t
+*/
+{
+    size_t index = 0;
+    int count = 0;
+    char *elements = list->elements;
+    while (index < list->size)
+    {
+        if (list->type == lst_list)
+        {
+            t_list *tmp = ((t_list**)elements)[index];
+            if (!(memcmp(item, tmp, sizeof(t_list))))
+                count += 1;
+        }
+        else if (!(memcmp(item, &(elements[index * tSizes[list->type]]), tSizes[list->type])))
+                count += 1;
+        index += 1;
+    }
+    return (count);
+}
+
+void     append(t_list* list, void* item)
+/*
+        Adds an element to the end of a t_list object
+
+        Parameters
+        ----------
+        list : t_list*
+            the t_list where to add the element
+
+        item : void*
+            the element to be added
+*/
+{
+    t_list listTmp;
+    listTmp.capacity = DEFAULT_CAP;
+    if (list->type == lst_list)
+        listTmp.elements = &item;
+    else
+        listTmp.elements = item;
+    listTmp.size = 1;
+    listTmp.type = list->type;
+    extend(list, &listTmp);
+}
+
+
+
+int   lst_index(t_list* list, void* item)
+/*
+        looks for the index of a specific element and returns it if it exists
+
+        Parameters
+        ----------
+        list : t_list*
+            the t_list where to search for the element
+
+        item : void*
+            the element to be indexed
+
+        Returns
+        =======
+        index : int
+*/
+{
+    size_t index = 0;
+    char *elements = (char*)(list->elements);
+    while (index < list->size)
+    {
+        if (list->type == lst_list)
+        {
+            t_list *tmp = ((t_list**)elements)[index];
+            t_list *itmTmp = (t_list*)item;
+            // NEEDS TO COMPARE LITERALS (group node items)
+            if (tmp->size == itmTmp->size || 
+                !(memcmp(itmTmp->elements, tmp->elements, tmp->size * tSizes[tmp->type]))) 
+                return index;
+        }
+        else if (!(memcmp(item, &elements[index * tSizes[list->type]], tSizes[list->type])))
+                return index;
+        index += 1;
+    }
+    return (-1);
+}
+
+size_t calculate_index(long int index, size_t size)
```

Comment:
I wonder if this function is necessary. For arrays and in Fortran we do this in the printing. As it is very simple this function would be better inlined anyway

Reply:
yeah I agree this could easily be calculated by pyccel

Reply:
What do you mean by "imported to python"? Are you trying to **replace** python's `list` object?

Reply:
my bad, for some reason I was under the impression that we might need to import the library alone and use these methods on pylistobjects